### PR TITLE
skip single selection for artist actions

### PR DESCRIPTION
**Pull request by aNNiMON**
Since most tracks are performed by a single artist, I found it a bit annoying to always select an artist when using the `GoToArtist` and `ShowActionsOnArtist` actions.

### CODE REVIEW

File: spotify_player/src/event/mod.rs [hunk 1]
```diff
@@ -144,11 +144,7 @@ pub fn handle_action_in_context(
                 }
             }
             Action::GoToArtist => {
-                ui.popup = Some(PopupState::ArtistList(
-                    ArtistPopupAction::Browse,
-                    track.artists,
-                    new_list_state(),
-                ));
+                self::handle_go_to_artist(track.artists, ui);
```
Comment (aome510) [inline spotify_player/src/event/mod.rs:147]:
```suggestion
                handle_go_to_artist(track.artists, ui);
```

File: spotify_player/src/event/mod.rs [hunk 2]
```diff
@@ -192,11 +188,7 @@ pub fn handle_action_in_context(
                 })?;
             }
             Action::ShowActionsOnArtist => {
-                ui.popup = Some(PopupState::ArtistList(
-                    ArtistPopupAction::ShowActions,
-                    track.artists,
-                    new_list_state(),
-                ));
+                self::handle_show_actions_on_artist(track.artists, data, ui)
```
Comment (aome510) [inline spotify_player/src/event/mod.rs:191]:
```suggestion
                handle_show_actions_on_artist(track.artists, data, ui)
```

File: spotify_player/src/event/mod.rs [hunk 3]
```diff
@@ -240,11 +228,7 @@ pub fn handle_action_in_context(
                 })?;
             }
             Action::ShowActionsOnArtist => {
-                ui.popup = Some(PopupState::ArtistList(
-                    ArtistPopupAction::ShowActions,
-                    album.artists,
-                    new_list_state(),
-                ));
+                self::handle_show_actions_on_artist(album.artists, data, ui);
```
Comment (aome510) [inline spotify_player/src/event/mod.rs:231]:
```suggestion
                handle_show_actions_on_artist(album.artists, data, ui);
```

File: spotify_player/src/event/mod.rs [hunk 4]
```diff
@@ -323,6 +307,43 @@ pub fn handle_action_in_context(
...
+    }
+}
+
+fn handle_show_actions_on_artist(
+    artists: Vec<Artist>,
+    data: &DataReadGuard,
+    ui: &mut UIStateGuard,
+) {
+    if artists.len() == 1 {
+        let actions = { construct_artist_actions(&artists[0], data) };
```
Comment (aome510) [inline spotify_player/src/event/mod.rs:333]:
```suggestion
        let actions = construct_artist_actions(&artists[0], data);
```

**Review by aome510 — APPROVED** [review]
Thanks for the nice QoL improvement. Few nits left.

File: spotify_player/src/event/mod.rs (hunk 3 above)
Comment (aNNiMON) [reply spotify_player/src/event/mod.rs]: Done. I wonder why CI doesn't detect it?

File: spotify_player/src/event/mod.rs (hunk 2 above)
Comment (aome510) [inline spotify_player/src/event/mod.rs:191]: you missed this one

Comment (aNNiMON) [reply spotify_player/src/event/mod.rs]: Oh, my bad 😄
Fixed

**Pull request merged**